cirq-core/cirq/work/sampler.py

```diff
@@ -334,34 +334,29 @@ def sample_expectation_values(
 
         # Flatten Circuit Sweep into one big list of Params.
         # Keep track of their indices so we can map back.
-        circuit_sweep = study.UnitSweep if params is None else study.to_sweep(params)
-        all_circuit_params: List[Dict[str, float]] = [
-            dict(circuit_params) for circuit_params in circuit_sweep.param_tuples()
-        ]
-        circuit_param_to_sweep_i: Dict[FrozenSet[str, float], int] = {
-            _hashable_param(param.items()): i for i, param in enumerate(all_circuit_params)
+        all_params: List[Dict[str, float]] = [pr.param_dict for pr in study.to_resolvers(params)]
+        circuit_param_to_sweep_i: Dict[FrozenSet[Tuple[str, float]], int] = {
+            _hashable_param(param.items()): i for i, param in enumerate(all_params)
         }
-        del params
 
-        accumulators = measure_observables(
+        obs_meas_results = measure_observables(
             circuit=program,
             observables=flat_pstrings,
             sampler=self,
             stopping_criteria=RepetitionsStoppingCriteria(total_repetitions=num_samples),
             readout_symmetrization=False,
-            circuit_sweep=circuit_sweep,
-            checkpoint=False,
+            circuit_sweep=params,
```

Comment:
To confirm: this is meant to be `params` (and not `all_params`), correct?

Reply:
I used to have better names, but it got a little messed up with the new flattening behavior. I'll try to change to better names. Should we stick with `params` as the sampler argument? Is that used in other run_sweep functions? If so, we're sortof stuck with that name.

Reply:
changing "all_params" to "flat_params"

Reply:
SGTM